### PR TITLE
TA 99049

### PR DESCRIPTION
**Pull request by katmayb**
Technical advisory around access to statement bundle diagnostics.

### CODE REVIEW

File: _includes/common/sql/statement-bundle-warning.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {{site.data.alerts.callout_danger}}Statement bundles can contain unredacted user data including histograms and placeholders that contain real data samples, statements that have real data which can contain PII information, and database schema that could be sensitive. Be careful when generating and sharing statement bundles.
 
 <br><br>To allow or disallow a role from seeing <a href="{{ link_prefix }}ui-statements-page.html#diagnostics">statements diagnostics bundles</a>, set the <code>VIEWACTIVITYREDACTED</code> <a href="{{ link_prefix }}create-role.html#role-options">role option</a>.
+
+<br><br>In CockroachDB v21.2.x, v22.1.0 to v22.1.16, v22.2.0 to v22.2.6, non-admin SQL users with an authenticated HTTP session could download statement diagnostic bundles given a bundle URL from the DB Console or <code>EXPLAIN ANALYZE (DEBUG)</code> statement with a valid HTTP session cookie. This has been resolved in v22.1.17 and v22.2.7. For more information, see the <a href="../advisories/a99049.html">Technical Advisory</a>.
```
Comment (andf-crl) [inline _includes/common/sql/statement-bundle-warning.md:5]: nit: `or the <code>EXPLAIN ANALYZE (DEBUG)</code> statement`

nit: I would put the TA# in the link, without which suggests linking to the list of all of them:

> see <a href="../advisories/a99049.html">Technical Advisory A99049</a>

File: advisories/a99049.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+affected_versions: v21.2.x, v22.1.0 to v22.1.16, v22.2.0 to v22.2.6
+advisory_date: 2023-03-27
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+This is a data leak/privilege escalation of statement diagnostic bundles by non-admin users. We expose the ability to download a statement diagnostic bundle over HTTP via a URL provided either by the [DB Console](../{{site.versions["stable"]}}/ui-statements-page.html#activate-diagnostics-collection-and-download-bundles) or the [SQL shell](../{{site.versions["stable"]}}/explain-analyze.html#debug-option). This URL is accessible to any logged-in user, but should be accessible only to admins. This bundle contains unredacted query execution data including query text and the values of variables used within.
```
Comment (andf-crl) [inline advisories/a99049.md:15]: Good opportunity to link `admin` here:

```
but should be accessible only to [admin users](security-reference/authorization.html#admin-role).
```

**Review by andf-crl — APPROVED** [review]
LGTM, just some minor nits. Thanks!

File: advisories/a99049.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+This is resolved in CockroachDB by PR [#99051](https://github.com/cockroachdb/cockroach/pull/99051), which fixes privilege escalation by correctly checking the SQL user’s privileges before granting access to the statement diagnostic bundle.
+
+The fix has been applied to maintenance releases of CockroachDB: [v22.1.17](../releases/v22.1.html) and [v22.2.7](../releases/v22.2.html).
+
+This public issue is tracked by [#99049](https://github.com/cockroachdb/cockroach/issues/99049).
+
+## Mitigation
+
+Users of CockroachDB v21.2, v22.1, and v22.2 are encouraged to upgrade to v22.1.17, v22.2.7, or a later version. They can alternatively block HTTP access to the statement diagnostics download endpoint on their cluster via an HTTP proxy.
```
Comment (dhartunian) [inline advisories/a99049.md:31]: Wonder if it's worth mentioning that if you never generate a bundle, or delete the bundles by dropping rows from `system.statement_diagnostics` to remove existing generated bundles you can also protect against any access.

**Review by dhartunian — APPROVED** [review]
LGTM. Just have one suggestion if you think it's worth including.

File: advisories/a99049.md (hunk 3 above)
Comment (kathancox) [reply advisories/a99049.md:31]: Something like this @dhartunian? I can add to the end of the Mitigation section?

> Users can also protect against access if they do not generate statement diagnostics bundles, or delete the bundles by dropping rows from `system.statement_diagnostics` to remove existing generated bundles.

**Pull request merged**